examples/statistics/time_series_histogram.py

```diff
@@ -0,0 +1,85 @@
+that are not immediately obvious.
+
+The first plot shows the typical way of visualizing multiple time series by
+overlaying them on top of each other with ``plt.plot``. The second and third
+plots show how to reinterpret the data as a 2d histogram, with optional
+interpolation.
+"""
+from copy import copy
+import time
+
+import numpy as np
+import numpy.matlib
+import matplotlib.pyplot as plt
+from matplotlib.colors import LogNorm
+
+_, axes = plt.subplots(nrows=3, figsize=(10, 6 * 3))
+
+# Make some data; a 1D random walk + small fraction of sine waves
+num_series = 1000
+num_points = 100
+SNR = 0.10  # Signal to Noise Ratio
+x = np.linspace(0, 4 * np.pi, num_points)
+# random walk
+Y = np.cumsum(np.random.randn(num_series, num_points), axis=-1)
+# sinusoidal signal
+num_signal = int(round(SNR * num_series))
+phi = (np.pi / 8) * np.random.randn(num_signal, 1)  # small random offest
+Y[-num_signal:] = (
+    np.sqrt(np.arange(num_points))[None, :]
+    * (np.sin(x[None, :] - phi)
+       + 0.05 * np.random.randn(num_signal, num_points))
+)
+
+# Plot it using `plot` and a small value of alpha. With this view it is
+# very difficult to observe the sinusoidal behavior because of how many
+# overlapping series there are. It also takes a bit of time to run because so
+# many individual artists that need to be generated.
+tic = time.time()
+axes[0].plot(x, Y.T, color="C0", alpha=0.1)
+toc = time.time()
+axes[0].set_title(
+    r"Standard time series visualization using line plot")
+print(f"{toc-tic:.2f} sec. elapsed")  # ~0.26 seconds
+
+
+# Now we will convert the multiple time series into a histogram. Not only will
+# the hidden signal be more visible, but it is also a much quicker procedure.
+tic = time.time()
+# linearly interpolate between the points in each time series
+num_fine = 800
+x_fine = np.linspace(x.min(), x.max(), num_fine)
+y_fine = np.empty((num_series, num_fine), dtype=float)
+for i in range(num_series):
+    y_fine[i, :] = np.interp(x_fine, x, Y[i, :])
+y_fine = y_fine.flatten()
+x_fine = np.matlib.repmat(x_fine, num_series, 1).flatten()
+
+
+# Plot (x, y) points in 2d histogram with log colorscale
+# It is pretty evident that there is some kind of structure under the noise
+# You can tune vmax to make signal more visible
+cmap = copy(plt.cm.plasma)
+cmap.set_bad(cmap(0))
+h, xedges, yedges = np.histogram2d(x_fine, y_fine, bins=[400, 100])
+axes[1].pcolormesh(xedges, yedges, h.T, cmap=cmap, norm=LogNorm(vmax=1.5e2))
+axes[1].set_title(
+    r"Alternative time series vis. using 2d histogram and log color scale")
+
+# Same thing on linear color scale but with different (more visible) cmap
+h, xedges, yedges = np.histogram2d(x_fine, y_fine, bins=[400, 100])
+axes[2].pcolormesh(xedges, yedges, h.T, cmap=cmap, vmax=1.5e2)
+axes[2].set_title(
+    r"Alternative time series vis. using 2d histogram and linear color scale")
+toc = time.time()
+# ~0.08 sec for both plots + interpolation
```

Comment:
```suggestion
```

The timings will show up in the rendered docs and will not match your timing, so lets not include confusing comments...